cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -362,9 +363,10 @@ findProjectRoot = do
 readProjectConfig :: Verbosity -> FilePath -> Rebuild ProjectConfig
 readProjectConfig verbosity projectRootDir = do
     global <- readGlobalConfig verbosity
-    local  <- readProjectLocalConfig      verbosity projectRootDir
-    extra  <- readProjectLocalExtraConfig verbosity projectRootDir
-    return (global <> local <> extra)
+    local  <- readProjectLocalConfig       verbosity projectRootDir
+    freeze <- readProjectLocalFreezeConfig verbosity projectRootDir
```

Comment:
Re-reading that ticket I think @tibbe's complains are handled, here are the main ones:

> I think this complicates things:
>
> * Users will need to understand what they should edit when and understand how the plethora of config files (.cabal, cabal.sandbox.config, cabal.config, cabal.freeze) interact.
> * The implementation is similarly complicated. Just to give an example among many: configure now needs to check if cabal.freeze has been updated and if so reconfigure.

By making the includes explicit it will be clear how the files interact: there's really just the cabal.project file but it can explicitly include other things. The new implementation already deals properly with tracking changes to the project files and reconfiguring correctly, so this isn't a problem.

I was just thinking about what the comment should be, e.g. we might want to explain that you can manage these manually by coping into the main file, but at the moment there's then no way to not include the generated one if it exists. So we may want to move forward on the explicit include approach so that we can provide a coherent story (including putting appropriate comments in the generated files).